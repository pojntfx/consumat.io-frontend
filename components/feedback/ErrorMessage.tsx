import { ExclamationCircleIcon, XIcon } from "@heroicons/react/solid";
import { useState } from "react";

type ErrorMessageProps = {
  message?: string;
};

const ErrorMessage = ({
  message = "Oops! Something went wrong. (≧Д≦)",
}: ErrorMessageProps) => {
  const [shouldBeDisplayed, setShouldBeDisplayed] = useState(true);

  return (
    shouldBeDisplayed && (
      <div className="flex justify-between bg-red-500 my-2 p-2 rounded shadow hover:shadow-md">
        <div className="flex justify-center items-center text-gray-50 font-semibold">
          <ExclamationCircleIcon className="h-6 w-6" />
          <p className="ml-2">{message}</p>
        </div>
        <button
          className="h-6 w-6 bg-transparent shadow-none hover:shadow-none text-gray-50"
          onClick={() => setShouldBeDisplayed(false)}
        >
          <XIcon />
        </button>
      </div>
    )
  );
};

export default ErrorMessage;
