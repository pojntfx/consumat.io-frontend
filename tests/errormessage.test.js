import React from "react";
import renderer from "react-test-renderer";
import ErrorMessage from "../components/feedback/ErrorMessage";

it("renders correctly", () => {
  const tree = renderer
    .create(<ErrorMessage message="An error occurred" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
