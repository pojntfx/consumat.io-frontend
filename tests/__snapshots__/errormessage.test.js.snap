// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-between bg-red-500 my-2 p-2 rounded shadow hover:shadow-md"
>
  <div
    className="flex justify-center items-center text-gray-50 font-semibold"
  >
    <svg
      className="h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fillRule="evenodd"
      />
    </svg>
    <p
      className="ml-2"
    >
      An error occurred
    </p>
  </div>
  <button
    className="h-6 w-6 bg-transparent shadow-none hover:shadow-none text-gray-50"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;
